docs/demography.md

```diff
@@ -1652,9 +1685,149 @@ distributed in space.
+and increases as the chance of them migrating to the same population increases.
+Interestingly, the coalescence rate of two lineages in population "B"
+is nonmonotonic, due to the asymmetric migration rates.
+
+Many properties of genomes (e.g., mean genetic diversity) are determined
+by mean coalescence times, which we can compute with
 {meth}`.DemographyDebugger.mean_coalescence_time`.
-:::
+For instance, we might want to know how the three pairwise mean coalescence times
+change as we make the migration rate less asymmetric:
+
+```{code-cell}
+def get_mean_coaltimes(m):
+    demography.set_migration_rate(source="A", dest="B", rate=m)
+    debug = demography.debug()
+    TAA = debug.mean_coalescence_time([2, 0])
+    TBB = debug.mean_coalescence_time([0, 2])
+    TAB = debug.mean_coalescence_time([1, 1])
+    return [TAA, TBB, TAB]
+
+M = np.linspace(0.001, 0.02, 11)
+T = np.zeros((len(M), 3))
+for j, m in enumerate(M):
+    T[j, :] = get_mean_coaltimes(m)
+
+plt.plot(M, T[:,0], label="two A samples")
+plt.plot(M, T[:,1], label="two B samples")
+plt.plot(M, T[:,2], label="one of each")
+plt.legend()
+plt.xlabel("A->B migration rate")
+plt.ylabel("coalecence rate");
+```
+
+From this, we can see that genetic diversity increases as the migration rate increases,
+that divergence between the populations is always greater than diversity within either,
+and that diversity within "B" goes from a level close to that found between populations
+to that within "A", as the "A->B" migration rate approaches the "B->A" rate.
+
+
+### Inverse instantaneous coalescence rates
+
+Coalescence rates are analogous to effective population sizes,
+in that if {math}`r(t)` is the coalescence rate at time {math}`t` ago
+between two lineages from a population in *any* population model,
+then a single population of changing size,
+with {math}`N(t) = 1/(2r(t))` diploids at time {math}`t` ago,
+has exactly the same distribution of coalescence times.
+One reason this is helpful is because we often fit simplified demographic models
+to more complex reality,
+and we might expect models that fit a single population-size-over-time
+to infer something closer to the inverse coalescence rate.
+For more discussion, see
+[Chikhi et al., 2018](https://doi.org/10.1038/s41437-017-0005-6) and
+[Adrion et al., 2020](https://doi.org/10.7554/eLife.54967).
+
+For instance, let's take the Gutenkunst et al 2009 "Out of Africa" model
+we used above in the {ref}`sec_demography_examples_population_tree` section,
+and compare inverse coalescence rates to actual population sizes:
+
+```{code-cell}
+:tags: [remove-cell]
+
+demography = msprime.Demography._ooa_model()
+```
+
+```{code-cell}
+# first define Gutenkunst et al model as above
+debug = demography.debug()
+
+T = np.concatenate([
+            np.linspace(0, 1000, 2001),
+            np.linspace(1000, 1e4, 401)[1:]
+])
+
+R = np.zeros((len(T), 3))
+R[:,0], _ = debug.coalescence_rate_trajectory(T, [2, 0, 0, 0, 0, 0])
```

Comment:
Hmm, so this indicates to me that the interface isn't great. Maybe we could should reuse the ``samples`` specification from ``sim_ancestry`` instead, so you can do things like ``{"A" : 2}`` rather than having to make an array of zeros? This'll get problematic for big models, where we've got 10s populations. Also seems like quite an error prone way of doing it.

Reply:
Yep - let's push that to a different issue, though?

Reply:
OK